tools/crm_resource_print.c

```diff
@@ -131,27 +131,22 @@ cli_resource_print(pe_resource_t *rsc, pe_working_set_t *data_set, bool expanded
     return pcmk_rc_ok;
 }
 
-PCMK__OUTPUT_ARGS("attribute-list", "pe_resource_t *", "char *", "GHashTable *")
+PCMK__OUTPUT_ARGS("attribute-list", "pe_resource_t *", "char *", "char *")
 static int
 attribute_list_default(pcmk__output_t *out, va_list args) {
     pe_resource_t *rsc = va_arg(args, pe_resource_t *);
     char *attr = va_arg(args, char *);
-    GHashTable *params = va_arg(args, GHashTable *);
+    char *value = va_arg(args, char *);
 
-    const char *value = NULL;
-
-    if (params != NULL) {
-        value = g_hash_table_lookup(params, attr);
-    }
     if (value != NULL) {
         out->begin_list(out, NULL, NULL, "Attributes");
         out->list_item(out, attr, "%s", value);
         out->end_list(out);
+        return pcmk_rc_ok;
     } else {
         out->err(out, "Attribute '%s' not found for '%s'", attr, rsc->id);
+        return pcmk_rc_unknown_format;
```

Comment:
It does make sense to return a different code for unknown attributes, but I wouldn't want to change the behavior now for backward compatibility reasons (same for text implementation). I think the idea behind the original code was that even if there is no such attribute, we successfully determined that.